Merge multiple parameters in newBPFFilter (#11535)
PowerDNS/pdns: pdns/dnsdistdist/docs/reference/ebpf.rst

  :param int seconds: The number of seconds this block to expire
  :param str msg: A message to display while inserting the block

.. function:: newBPFFilter(options) -> BPFFilter

  .. versionchanged:: 1.7.0
    This function now supports a table for each parameters, and the ability to use pinned eBPF maps.
  .. versionchanged:: 1.8.0
    This function now get the parameters via a table.

  Return a new eBPF socket filter with a maximum of maxV4 IPv4, maxV6 IPv6 and maxQNames qname entries in the block tables.
  Maps can be pinned to a filesystem path, which makes their content persistent across restarts and allows external programs to read their content and to add new entries. dnsdist will try to load maps that are pinned to a filesystem path on startups, inheriting any existing entries, and fall back to creating them if they do not exist yet. Note that the user dnsdist is running under must have the right privileges to read and write to the given file, and to go through all the directories in the path leading to that file. The pinned path must be on a filesystem of type ``BPF``, usually below ``/sys/fs/bpf/``.

>   :param table options: A table with key: value pairs with webserver options.

Reviewer: ```suggestion
  :param table options: A table with key: value pairs with options.
```

  Options:

>   * ``v4Params``: int or table - Maximum number of entries in this filter or a table with ``maxItems`` or ``pinnedPath`` .

Reviewer: I wonder if it wouldn't be easier for the three "*Params" to always take a table and drop the `int` option that can be specified via `maxItems` anyway.

Author: > I wonder if it wouldn't be easier for the three "*Params" to always take a table and drop the `int` option that can be specified via `maxItems` anyway.

Yes, I think that's a good idea. I will do that.

  :param int seconds: The number of seconds this block to expire
  :param str msg: A message to display while inserting the block

.. function:: newBPFFilter(options) -> BPFFilter
              newBPFFilter(v4Parameters, v6Parameters, qnamesParameters) -> BPFFilter (1.7.x)
              newBPFFilter(maxV4, maxV6, maxQNames) -> BPFFilter (before 1.7.0)

  .. versionchanged:: 1.7.0
    This function now supports a table for each parameters, and the ability to use pinned eBPF maps.
  .. versionchanged:: 1.8.0

>     This function now get the parameters via a table.

Reviewer: ```suggestion
    This function now gets its parameters via a table.
```
